packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts

```diff
@@ -1,29 +1,35 @@
 import { defaultCompareMediaQueries, GriffelRenderer } from '@griffel/core';
 import { Compilation } from 'webpack';
-import type { Compiler, sources } from 'webpack';
+import type { Chunk, Compiler, Module, sources } from 'webpack';
 
-import { sortCSSRules } from './sortCSSRules';
 import { parseCSSRules } from './parseCSSRules';
+import { sortCSSRules } from './sortCSSRules';
 
 export type GriffelCSSExtractionPluginOptions = {
   compareMediaQueries?: GriffelRenderer['compareMediaQueries'];
 };
 
-/**
- * Forces all files with `griffel.css` be concatenated into a single asset.
- */
-function forceCSSIntoOneStyleSheet(compiler: Compiler) {
-  compiler.options.optimization ??= {};
-  compiler.options.optimization.splitChunks ??= {};
-  compiler.options.optimization.splitChunks.cacheGroups ??= {};
-
-  compiler.options.optimization.splitChunks.cacheGroups['griffel'] = {
-    name: `griffel`,
-    type: 'css/mini-extract',
-    chunks: 'all',
-    test: /griffel.css/,
-    enforce: true,
-  };
+type IterableElement<TargetIterable> = TargetIterable extends Iterable<infer ElementType> ? ElementType : never;
+type ChunkGroup = IterableElement<Chunk['groupsIterable']>;
+
+const PLUGIN_NAME = 'GriffelExtractPlugin';
+
+function attachGriffelChunkToMainEntryPoint(compilation: Compilation, griffelChunk: Chunk) {
+  const entryPoints = Array.from(compilation.entrypoints.values());
+
+  if (entryPoints.length === 0) {
+    throw new Error('Failed to find and entry points in "compilation.entrypoints"');
+  }
+
+  const mainEntryPoint = entryPoints[0];
```

Comment:
is the main entrypoint guaranteed to be the first entrypoint in `compilation.entrypoints` ?

Reply:
actually, what if webpack is configured with multiple entrypoints?

Reply:
It's a good question. In all scenarios that I currently tested attaching to main entrypoint was enough.